passkey-client/src/client_data.rs

```diff
@@ -0,0 +1,54 @@
+use serde::Serialize;
+
+/// A trait describing how client data should be generated during a WebAuthn operation.
+pub trait ClientData<E: Serialize> {
+    /// Extra client data to be appended to the automatically generated client data.
+    fn extra_client_data(&self) -> E;
+
+    /// The hash of the client data to be used in the WebAuthn operation.
+    fn client_data_hash(&self) -> Option<Vec<u8>>;
+}
```

Comment:
**Praise:** I really like this pattern

Reply:
Thank you! To be honest this is the part I thought had the biggest risk of having to be rewritten, glad you liked it!

Reply:
Thats really impressive!